src/main/osd/osd.c

```diff
@@ -1046,6 +1046,9 @@ void osdUpdate(timeUs_t currentTimeUs)
 
     if (!osdIsReady) {
         if (!displayCheckReady(osdDisplayPort, false)) {
+            if (osdDisplayPortDeviceType == OSD_DISPLAYPORT_DEVICE_FRSKYOSD) {
+                displayRedraw(osdDisplayPort);
```

Comment:
Interesting find. What is the reason that this is needed here? Maybe there should be a comment as to why we need to redraw when the FrSkyOSD device isn't ready.

Reply:
Agreed, this is properly weird.  To me it feels like this should be handled in the initialisation code as the 'seach for MAX7456' should only happen once.